coriolis/tests/test_events.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2023 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+import logging
+from unittest import mock
+
+import ddt
+
+from coriolis import constants
+from coriolis import events
+from coriolis.tests import test_base
+
+
+@ddt.ddt
+class EventManagerTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis EventManager class."""
+
+    def setUp(self):
+        super(EventManagerTestCase, self).setUp()
+        self.mock_event_handler = mock.Mock()
+        self.event_manager = events.EventManager(self.mock_event_handler)
+
+    def test_call_event_handler(self):
+        mock_method = mock.Mock()
+        self.mock_event_handler.test_method = mock_method
+
+        result = events.EventManager._call_event_handler(
+            self.event_manager, 'test_method', 'arg1', 'arg2', kwarg1='value1')
+
+        mock_method.assert_called_once_with('arg1', 'arg2', kwarg1='value1')
+        self.assertEqual(result, mock_method.return_value)
+
+    def test_call_event_handler_method_not_exist(self):
+        mock_event_handler = mock.Mock(spec_set=[])
+        event_manager = events.EventManager(mock_event_handler)
+
+        self.assertRaises(AttributeError, event_manager._call_event_handler,
+                          'non_existent_method', 'arg1', 'arg2',
+                          kwarg1='value1')
+
+    def test_call_event_handler_no_event_handler(self):
+        self.event_manager._event_handler = None
+        result = self.event_manager._call_event_handler('test_method')
+        self.assertEqual(result, None)
+
+    @ddt.data(
+        {'message': 'test-message',
+         'total_steps': -10,
+         'initial_step': 0,
+         'expected_log_level': logging.WARN},
+        {'message': 'test-message',
+         'total_steps': 0,
+         'initial_step': 0,
+         'expected_log_level': logging.WARN},
+        {'message': 'test-message',
+         'total_steps': 50,
+         'initial_step': 100,
+         'expected_exception': ValueError},
+        {'message': 'test-message',
+         'total_steps': 100,
+         'initial_step': 100,
+         'expected_perc': 100,
+         'expected_initial_step': 100,
+         'expected_total_steps': 100},
+        {'message': 'test-message',
+         'total_steps': 100,
+         'initial_step': 50,
+         'expected_perc': 50,
+         'expected_initial_step': 50,
+         'expected_total_steps': 100})
```

Comment:
Please add a case where `expected_perc` remains 0, i.e. this `if` is skipped:
https://github.com/cloudbase/coriolis/blob/7b1e26f736b8f6a5f7cfb3cf6cfb4dcc728b2657/coriolis/events.py#L55-L57